### DISCUSSION TITLE
"WeakMap" caching semantics

### DISCUSSION THREAD

**Discussion by chrisvasz** [TanStack/query#5109] [Ideas]
I'm using react-query to store the results of an expensive data transformation. I split the calculation into batches to avoid freezing the main thread, which makes the calculation asynchronous. I want the cached value inside react-query to expire not on a timer, but via WeakMap-like semantics. If the original data is GCed, then I want the cached transformation to be subject to GC also. Currently, I'm using a small value for `cacheTime`, but that solution is imperfect because the data may actually need to stay cached for a while – it totally depends on how the user navigates the UI.

Is there a way to expire data in react-query not based on time, but based on WeakMap-like semantics?

**Comment by TkDodo** [comment]
no I don't think that's possible
